llvm/docs/InstrProfileFormat.rst

```diff
@@ -0,0 +1,478 @@
+            |   |   |    |  +-----------------------+---+
+            |   |   |    |  |   Profile Summary     |   |
+            |   |   |    |  +-----------------------+   P
+            |   |   +------>|  Function PGO data    |   A
+            |   |        |  +-----------------------+   Y
+            |   +---------->|  MemProf profile data |   L
+            |            |  +-----------------------+   O
+            |            +->|    Binary Ids         |   A
+            |               +-----------------------+   D
+            +-------------->|  Temporal profiles    |   |
+                            +-----------------------+---+
+
+Header
+--------
+
+``Magic``
+  The purpose of the magic number is to be able to tell if the profile is an
+  indexed profile.
+
+``Version``
+  Similar to raw profile version, the lower 32 bits specifies the version of the
+  indexed profile and the most significant 32 bits are reserved to specify the
+  variant types of the profile.
+
+``HashType``
+  The hashing scheme for on-disk hash table keys. Only MD5 hashing is used as of
+  writing.
+
+``HashOffset``
+  An on-disk hash table stores the per-function profile records. This field records
+  the offset of this hash table's metadata (i.e., the number of buckets and
+  entries), which follows right after the payload of the entire hash table.
+
+``MemProfOffset``
+  Records the byte offset of MemProf profiling data.
+
+``BinaryIdOffset``
+  Records the byte offset of binary id sections.
+
+``TemporalProfTracesOffset``
+  Records the byte offset of temporal profiles.
+
+Payload Sections
+------------------
+
+(CS) Profile Summary
+^^^^^^^^^^^^^^^^^^^^^
+This section is right after profile header. It stores the serialized profile
+summary. For context-sensitive IR-based instrumentation PGO, this section stores
+an additional profile summary corresponding to the context-sensitive profiles.
+
+Function PGO data
+^^^^^^^^^^^^^^^^^^
+This section stores functions and their PGO profiling data as an on-disk hash
+table. Profile data for functions with the same name are grouped together and
+share one hash table entry (the functions may come from different shared libraries
+for instance). The profile data for them are organized as a sequence of key-value
+pair where the key is `FuncHash`_, and the value is profiled information (represented
+by `InstrProfRecord`_) for the function.
+
+.. _`InstrProfRecord`: https://github.com/llvm/llvm-project/blob/7e405eb722e40c79b7726201d0f76b5dab34ba0f/llvm/include/llvm/ProfileData/InstrProf.h#L693
+
+MemProf Profile data
+^^^^^^^^^^^^^^^^^^^^^^
+This section stores function's memory profiling data. See
+`MemProf binary serialization format RFC`_ for the design.
+
+.. _`MemProf binary serialization format RFC`: https://lists.llvm.org/pipermail/llvm-dev/2021-September/153007.html
+
+Binary Ids
+^^^^^^^^^^^^^^^^^^^^^^
+The section is used to carry on binary-id information from raw profiles.
```

Comment:
Add a link to the RFC: https://discourse.llvm.org/t/rfc-adding-binary-id-into-llvm-profiles/58465

Reply:
done.